File: Frontend/GoticketFrontend/src/services/eventService.js
import api from "./api";

const API_URL = "/events";

export const createEvent = async (eventData) => {
  const response = await api.post(API_URL, eventData);
  return response.data;
};

export const updateEvent = async (id, eventData) => {
  const response = await api.put(`${API_URL}/${id}`, eventData);
  return response.data;
};

export const deleteEvent = async (id) => {
  await api.delete(`${API_URL}/${id}`);
};

export const getAllEvents = async () => {
  const response = await api.get(API_URL);
  return response.data;
};

export const getEventById = async (id) => {
  const response = await api.get(`${API_URL}/${id}`);
  return response.data;
};

export const getSeatsForEvent = async (eventId) => {
  const response = await api.get(`${API_URL}/${eventId}/seats`);
  return response.data;
};

export const getAvailableSeatsForEvent = async (eventId) => {
  const response = await api.get(`${API_URL}/${eventId}/seats/available`);
  return response.data;
};

File: Backend/GoticketBackend/src/main/java/com/goticket/service/EventService.java
package com.goticket.service;

import com.goticket.model.Event;
import com.goticket.model.Seat;

import java.util.List;
import java.util.Optional;

public interface EventService {
    Event createEvent(Event event);
    Event updateEvent(Long id, Event event);
    void deleteEvent(Long id);

    List<Event> getAllEvents();
    Optional<Event> getEventById(Long id);

    List<Seat> getSeatsForEvent(Long eventId);
    List<Seat> getAvailableSeatsForEvent(Long eventId);
}
